### PR TITLE
[Refactor] Add shouldHideAvailability property to ZMUser

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

Moved the `shouldHideAvailability` property implementation & tests from `wire-ios` to `wire-ios-data-model`

### CODE REVIEW

**Comment by zenkins** [comment]
Can one of the admins verify this patch?

File: Tests/Source/Model/User/UserAvailabilityTests.swift [hunk 1]
```diff
@@ -0,0 +1,102 @@
+//
+//  UserAvailabilityTests.swift
+//  WireDataModelTests
+//
+//  Created by David Henner on 25.11.19.
+//  Copyright © 2019 Wire Swiss GmbH. All rights reserved.
+//
```
Comment (billypchan) [inline Tests/Source/Model/User/UserAvailabilityTests.swift]: boilerplate should be copied.

Comment (David-Henner) [reply Tests/Source/Model/User/UserAvailabilityTests.swift]: What do you mean @billypchan ?

Comment (David-Henner) [reply Tests/Source/Model/User/UserAvailabilityTests.swift]: Ah nevermind I got it

**Comment by David-Henner** [comment]
@zenkins release major

**Comment by billypchan** [comment]
@zenkins release major

**Pull request merged**